src/base/fm-module.c

```diff
@@ -527,7 +527,7 @@ gboolean fm_modules_add_directory(const char *path)
     GSList *l;
     gboolean res = FALSE;
 
-    g_return_val_if_fail(path != NULL || path[0] != '/', FALSE);
+    g_return_val_if_fail(path == NULL || path[0] != '/', FALSE);
```

Comment:
The test was incorrect but the change is incorrect too.
Correct assert condition should be **path != NULL && path[0] == '/'**.
Thank you very much for noticing this.